docs/arcadia-users-group/20230214-code-review-for-pubs/lesson.md

```diff
@@ -0,0 +1,88 @@
+
+**Question:** Could code review happen after version 1 of the pub?
+
+**Thoughts:** The goal of code review in the pub release process is to make sure someone inside or outside of Arcadia could pick up the code you used and with minimal effort, reproduce the results you got and reuse the code you wrote.
+We think that code review prior to the release of the first version of a pub will increase the usefulness of the computational products we're releasing to the world.
+
+Our goal is to have the code review component of the pub release process become a formality; we want code review to happen quickly after a unit of code is written (a new function, a new analysis notebook, a new figure) so that the feedback you get is actually useful and doesn't slow you down.
+If we can get to that point, the checks in the pub release cycle will be super quick.
+To make this process faster and to reduce the heat around version 1 pub releases, we've created [a checklist](#minimums-for-passing-code-review-a-checklist) of the five things that are needed to approve code associated with the first release of a pub.
+Any other suggestions you get are just that – suggestions – and can be punted to later pub releases or ignored.
+
+## Overview of the pub process and how it relates to code review
+
+![](overview.png)
+
+Checkout the **What to expect from code review during the pub process** Notion page for an up to date guide through this process.
+
+## Minimums for passing code review: a checklist
+
+1. All software packages and their versions are documented.
+2. Data inputs and outputs are documented.
+3. Only relative paths (not absolute paths) are used and the relative paths reference files in the repo or there is documentation for how to get the file.
+4. Enough comments and/or documentation are provided so it's clear what the code does.
+5. DOI for pub is linked in the README.
+
+## Most common pain points with code review
+
+1. Pushing to `main` or `master` instead of to a branch and opening a pull request
+2. Requesting review of the entire repo instead of requesting periodic reviews
+
+## GitHub refresher
+
+<center>
+<figure markdown>
+  ![Image title](../../workshops/20220920-intro-to-git-and-github/allison_horst_git_overview.png){ width="900" }
+  <figcaption> Figure by Allison Horst. <a href='https://twitter.com/allison_horst/status/1563210538510737409?s=20&t=USB46onUf9i7zYnkqjUSPQ' target='_blank'>www.twitter.com/allison_horst</a> </figcaption>
+</figure>
+</center>
+
+1. Make a new folder wherever your GitHub repos live. Name it `tmp-yourinitials-aug`.
+1. Download [this file](https://raw.githubusercontent.com/Arcadia-Science/2022-prjna853785-sourmash/main/scripts/utils.R) as our pretend code that we've been working on. Save it in your new folder.
+2. Create an empty, private repository in the Arcadia-Science organization. Use the same name as the local folder we made.
```

Comment:
Let's maybe create a private demo repo and make people open PRs there. We don't want to clean up 30 (potentially) unused repos.

Reply:
oh that's a great idea. i'll change this!